azkaban-db/src/main/java/azkaban/db/MySQLDataSource.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.db;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.apache.log4j.Logger;
+
+
+public class MySQLDataSource extends AzkabanDataSource {
+
+  private static Logger logger = Logger.getLogger(MySQLDataSource.class);
+
+  private static volatile MySQLDataSource instance = null;
+
+  private MySQLDataSource(String host, int port, String dbName,
+      String user, String password, int numConnections) {
+    super();
+
+    String url = "jdbc:mysql://" + (host + ":" + port + "/" + dbName);
+    addConnectionProperty("useUnicode", "yes");
+    addConnectionProperty("characterEncoding", "UTF-8");
+    setDriverClassName("com.mysql.jdbc.Driver");
+    setUsername(user);
+    setPassword(password);
+    setUrl(url);
+    setMaxTotal(numConnections);
+    setValidationQuery("/* ping */ select 1");
+    setTestOnBorrow(true);
+  }
+
+  /**
+   * Get a singleton object for MySQL BasicDataSource
+   */
+  public static MySQLDataSource getInstance(String host, int port, String dbName,
+      String user, String password, int numConnections) {
+    if (instance == null) {
+      synchronized (MySQLDataSource.class) {
+        if (instance == null) {
+          logger.info("Instantiating MetricReportManager");
+          instance = new MySQLDataSource(host, port, dbName, user, password, numConnections);
+        }
+      }
+    }
+    return instance;
+  }
+
+  /**
+   * This method overrides {@link BasicDataSource#getConnection()}, in order to have retry logics.
+   *
+   */
+  @Override
+  public synchronized Connection getConnection() throws SQLException {
+
+      /*
```

Comment:
This style was recommended by intellij for some reason. Just googled. every line of comment should start by `*`.

Modified.